sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -316,168 +310,161 @@ static int register_device()
   pubmsg.qos = 1;
   pubmsg.retained = 0;
 
-  // Send the registration request message
-  if ((rc = MQTTClient_publishMessage(mqtt_client, register_publish_topic, &pubmsg, NULL))
+  if ((rc = MQTTClient_publishMessage(mqtt_client, register_publish_topic_buffer, &pubmsg, NULL))
       != MQTTCLIENT_SUCCESS)
   {
-    (void)printf("Failed to publish register request: MQTTClient return code %d .\n", rc);
-    return rc;
+    LOG_ERROR("Failed to publish register request: MQTTClient return code %d .\n", rc);
+    exit(rc);
   }
 
-  return MQTTCLIENT_SUCCESS;
+  return;
 }
 
-static int get_operation_status()
+static void receive_registration_status()
 {
   int rc;
-  bool is_operation_complete = false;
   char* topic = NULL;
   int topic_len = 0;
   MQTTClient_message* message = NULL;
   az_iot_provisioning_client_register_response response;
   az_iot_provisioning_client_operation_status operation_status;
+  bool is_operation_complete = false;
 
-  // Continue to parse incoming responses from the Provisioning Service until
+  // Continue to parse incoming responses from the Device Provisioning Service until
   // the device has been successfully provisioned or an error occurs
   do
   {
-    // Receive message.
-    // If MQTTClient_receive is successful, topic and message memory must later be freed
-    if ((rc
-         = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
-        != MQTTCLIENT_SUCCESS)
+    if (((rc
+          = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
+         != MQTTCLIENT_SUCCESS)
+        && (MQTTCLIENT_TOPICNAME_TRUNCATED != rc))
     {
-      (void)printf("Failed to receive message: MQTTClient return code %d .\n", rc);
-      return rc;
+      LOG_ERROR("Failed to receive message: MQTTClient return code %d .\n", rc);
+      exit(rc);
     }
-    (void)printf("Received a message from service.\n");
-
-    // Parse operation message
-    if (az_failed(
-            rc = parse_operation_message(topic, topic_len, message, &response, &operation_status)))
+    else if (!message)
```

Comment:
this kind of implies that message is a bool

```suggestion
    else if (message == NULL)
```

Reply:
haha fine. i'm amused to see the different approaches to this... my last team I had a senior developer who would ONLY use the first style to check if a pointer was NULL or not.  The thought was stating "if (message == NULL)" is unnecessary since you can do it in a simpler way.